File: CsChatClient/Models/UnreadCount.cs
using System;
using Newtonsoft.Json;


namespace CsChatClient.Models
{
    public class UnreadCount : ISerializable
    {
        /// <summary>
        /// 单聊未读数
        /// </summary>
        public int Unread { get; set; }
        /// <summary>
        /// 群聊@数
        /// </summary>
        public int UnreadMention { get; set; }
        /// <summary>
        /// 群聊@All数
        /// </summary>
        public int UnreadMentionAll { get; set; }

        public void Serialize(JsonWriter writer)
        {
            throw new NotImplementedException();
        }

        public bool Unserialize(JsonReader reader)
        {
            while (reader.Read())
            {
                switch (reader.TokenType)
                {
                    case JsonToken.PropertyName:
                        if (reader.Value.Equals("unread"))
                        {
                            Unread = JsonTools.GetNextInt(reader);
                        }
                        else if (reader.Value.Equals("unreadMention"))
                        {
                            UnreadMention = JsonTools.GetNextInt(reader);
                        }
                        else if (reader.Value.Equals("unreadMentionAll"))
                        {
                            UnreadMentionAll = JsonTools.GetNextInt(reader);
                        }
                        break;
                    case JsonToken.EndObject:
                        return true;
                }
            }
            return false;
        }
    }
}

File: CppChatClient/model/UnreadCount.cpp
//
//  UnreadCount.cpp
//  ChatClient
//
//  Created by Hao Jia on 2020/1/30.
//  Copyright © 2020 WF Chat. All rights reserved.
//

#include "UnreadCount.h"
#include "../JsonTools.h"

namespace WFCLib {
void UnreadCount::Unserialize(const void *pvalue) {
    const Value& value = *((const Value *)pvalue);
    if (!value.IsObject()) {
        return;
    }

//            int unread;
    getValue(value, "unread", unread);
//            int unreadMention;
    getValue(value, "unreadMention", unreadMention);
//            int unreadMentionAll;
    getValue(value, "unreadMentionAll", unreadMentionAll);
}

void UnreadCount::Serialize(void *pwriter) const {
    Writer<StringBuffer> &writer = *((Writer<StringBuffer> *)pwriter);

    writer.StartObject();

    writer.String("unread");
    writer.Int(unread);
    writer.String("unreadMention");
    writer.Int(unreadMention);
    writer.String("unreadMentionAll");
    writer.Int(unreadMentionAll);

    writer.EndObject();
}
};
